src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -76,10 +73,9 @@ public Task Init(Func<MessageContext, Task> onMessage, Func<ErrorContext, Task<E
         public void Start(PushRuntimeSettings limitations)
         {
             maxConcurrency = limitations.MaxConcurrency;
-            semaphore = new SemaphoreSlim(limitations.MaxConcurrency, limitations.MaxConcurrency);
```

Comment:
Technically we could still use a semaphore but since the client now takes care of the concurrency I replaced it with a simple tracker for the shutdown aspects of the code